theseus/geometry/manifold.py

```diff
@@ -40,7 +41,12 @@ def __init__(
         if tensor is None and dtype is None:
             dtype = torch.get_default_dtype()
         if tensor is not None:
-            tensor = self._check_tensor(tensor, strict)
+            if not _FunctorchContext.get_context():
+                tensor = self._check_tensor(tensor, strict)
+            else:
+                warnings.warn(
+                    "functorch is enabled and tensor is not checked for validness."
```

Comment:
How does a user 'enable' functorch use. Does this happen under the hood when you use for instance vmap from functorch? We should make the warning more descriptive. Also 'tensor' -> 'Manifold type'?

Reply:
This warning also might be too spammy if we raise this for every new Manifold object. Is there a way to raise this only once (or for the user to turn it off)?

Reply:
The `warnings` package suppresses repetitions from the same source. 

<img width="809" alt="image" src="https://user-images.githubusercontent.com/4759586/186931732-f3cb9b57-09bf-42ba-832e-a30aa5427523.png">

Reply:
BTW, we should add a warning category. I think `RuntimeWarning` is appropriate.

Reply:
Oh that's cool! Agreed on adding the category.

Reply:
Refactored.